src/univariate/discrete/discretenonparametric.jl

```diff
@@ -312,3 +312,9 @@ end
 fit_mle(::Type{<:DiscreteNonParametric},
         ss::DiscreteNonParametricStats{T,W,Ts,Ws}) where {T,W,Ts,Ws} =
     DiscreteNonParametric{T,W,Ts,Ws}(ss.support, normalize!(copy(ss.freq), 1), check_args=false)
+
+# Collect a MixtureModel of DiscreteNonParametrics
+function DiscreteNonParametric(mm::MixtureModel{<:VariateForm,Discrete,DiscreteNonParametric{T,P,Ts,Ps}}) where {T,P,Ts,Ps}
```

Comment:
I wonder if it would be more appropriate to define a `convert` method than a constructor in this case (https://docs.julialang.org/en/v1/manual/conversion-and-promotion/#Conversion-vs.-Construction).

In any case, I think probably we could make it less restrictive:
```suggestion
function DiscreteNonParametric(mm::UnivariateMixture{Discrete,<:DiscreteNonParametric)
```

Reply:
Regarding `convert` - tell me whatever you think is best. `convert` makes sense although to me it seems a bit less convenient, as you need to specify the type `DiscreteNonParametric` although there is no really any other option that I can think of. Initially to me it felt like a `collect` but eventually I wrote it as a constructor.